cardea-server/src/main/java/ca/on/oicr/gsi/cardea/server/service/CaseService.java

```diff
@@ -179,14 +179,18 @@ private ShesmuSequencing makeShesmuSequencing(List<Sample> samples, MetricCatego
     for (Sample sample : samples) {
       if (sample.getRun() != null) {
 
-        if (sample.getDataReviewPassed() == null || (sample.getQcPassed() == null && sample.getQcUser() == null)) {
+        if (sample.getDataReviewPassed() == null
+            || (sample.getQcPassed() == null && sample.getQcUser() == null)) {
           hasWaiting = true;
-        } else if (Boolean.TRUE.equals(sample.getDataReviewPassed()) && Boolean.TRUE.equals(sample.getQcPassed())) {
+        } else if (isTrue(sample.getDataReviewPassed()) && isTrue(sample.getQcPassed())
+            && isTrue(sample.getRun().getQcPassed())
+            && isTrue(sample.getRun().getDataReviewPassed())) {
```

Comment:
Is this necessary? For our SOPs, if the run's QC is set to failed then the run-library's QC is set to failed. If the run's QC is unset, the lab shouldn't yet go on to QC the run-libraries.

Reply:
It's only a pass after the run and run-library sign-offs are both done. SOP may dictate the order in which those are _supposed_ to happen, but can't guarantee it, and checking 2 Booleans doesn't cost us anything.